sql/src/test/java/io/crate/integrationtests/CopyIntegrationTest.java

```diff
@@ -93,6 +93,60 @@ public void testCopyFromFile() throws Exception {
         assertThat(response.rows()[0][0], is("Don't pa\u00f1ic."));
     }
 
+    @Test
+    public void testCopyFromFileWithCSVExtension() throws Exception {
+        execute("create table quotes (id int primary key, " +
+            "quote string index using fulltext) with (number_of_replicas = 0)");
+        ensureYellow();
+
+        execute("copy quotes from ?", new Object[]{copyFilePath + "test_copy_from.csv"});
+        assertEquals(3L, response.rowCount());
+        refresh();
+
+        execute("select * from quotes");
+        assertEquals(3L, response.rowCount());
+        assertThat(response.rows()[0].length, is(2));
+
+        execute("select quote from quotes where id = 1");
+        assertThat(response.rows()[0][0], is("Don't pa\u00f1ic."));
+    }
+
+    @Test
+    public void testCopyFromFileWithCSVOption() {
+        execute("create table quotes (id int primary key, " +
+            "quote string index using fulltext) with (number_of_replicas = 0)");
+        ensureYellow();
+
+        execute("copy quotes from ? with (format='csv')", new Object[]{copyFilePath + "test_copy_from_csv.ext"});
+        assertEquals(3L, response.rowCount());
+        refresh();
+
+        execute("select * from quotes");
+        assertEquals(3L, response.rowCount());
+        assertThat(response.rows()[0].length, is(2));
+
+        execute("select quote from quotes where id = 1");
+        assertThat(response.rows()[0][0], is("Don't pa\u00f1ic."));
+    }
+
+    @Test
+    public void testCopyFromFileWithoutOptionOrRelevantFileExtension() throws Exception {
```

Comment:
Do we need 3 full blown integration tests to verify all variants of the format handling? I think covering this with unit tests and having 1 test that verifies the interaction would be sufficient (that could even be on a unit test level as well - e.g. on the level of the `FileReadingIterator`.

We don't need to test the "insert logic" here at all.

Reply:
Yeah, that's true. These tests were probably of more use to me during development, and three is most likely excessive given that it's already tested quite thoroughly in the unit tests. I still think it's a good idea to have one fully end-to-end test here though.